lib/Service/RecentPagesService.php

```diff
@@ -0,0 +1,127 @@
+
+use OCA\Collectives\Model\PageInfo;
+use OCA\Collectives\Model\RecentPage;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\IMimeTypeLoader;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IURLGenerator;
+use OCP\IUser;
+
+class RecentPagesService {
+
+	protected CollectiveService $collectiveService;
+	protected IURLGenerator $urlGenerator;
+	protected IDBConnection $dbc;
+	protected IConfig $config;
+	protected IMimeTypeLoader $mimeTypeLoader;
+
+	public function __construct(
+		CollectiveService $collectiveService,
+		IDBConnection $dbc,
+		IConfig $config,
+		IMimeTypeLoader $mimeTypeLoader,
+		IURLGenerator $urlGenerator
+	) {
+		$this->mimeTypeLoader = $mimeTypeLoader;
+		$this->config = $config;
+		$this->dbc = $dbc;
+		$this->urlGenerator = $urlGenerator;
+		$this->collectiveService = $collectiveService;
+	}
+
+	/**
+	 * @return RecentPage[]
+	 * @throws MissingDependencyException
+	 * @throws Exception
+	 */
+	public function forUser(IUser $user, int $limit = 10): array {
+		try {
+			$collectives = $this->collectiveService->getCollectives($user->getUID());
+		} catch (NotFoundException|NotPermittedException $e) {
+			return [];
+		}
+
+		$qb = $this->dbc->getQueryBuilder();
+		$appData = $this->getAppDataFolderName();
+		$mimeTypeMd = $this->mimeTypeLoader->getId('text/markdown');
+
+		$expressions = [];
+		$collectivesMap = [];
+		foreach ($collectives as $collective) {
+			$value = sprintf($appData . '/collectives/%d/%%', $collective->getId());
+			$expressions[] = $qb->expr()->like('f.path', $qb->createNamedParameter($value, IQueryBuilder::PARAM_STR));
+			$collectivesMap[$collective->getId()] = $collective;
+		}
+		unset($collectives);
+
+		$qb->select('p.*', 'f.mtime as timestamp', 'f.name as filename', 'f.path as path')
+			->from('filecache', 'f')
+			->leftJoin('f', 'collectives_pages', 'p', $qb->expr()->eq('f.fileid', 'p.file_id'))
+			->where($qb->expr()->orX(...$expressions))
+			->andWhere($qb->expr()->eq('f.mimetype', $qb->createNamedParameter($mimeTypeMd, IQueryBuilder::PARAM_INT)))
+			->orderBy('f.mtime', 'DESC')
+			->setMaxResults($limit);
+
+		$r = $qb->executeQuery();
+
+		$pages = [];
+		while ($row = $r->fetch()) {
+			$collectiveId = (int)explode('/', $row['path'], 4)[2];
+			if (!isset($collectivesMap[$collectiveId])) {
+				continue;
+			}
+
+			// cut out $appDataDir/collectives/%d/ prefix from front, and filename at the rear
+			$splitPath = explode('/', $row['path'], 4);
+			$internalPath = dirname(array_pop($splitPath));
+			unset($splitPath);
+
+			// prepare link and title
+			$pathParts = [$collectivesMap[$collectiveId]->getName()];
+			if ($internalPath !== '' && $internalPath !== '.') {
+				$pathParts = array_merge($pathParts, explode('/', $internalPath));
+			}
+			if ($row['filename'] !== 'Readme.md') {
+				$pathParts[] = basename($row['filename'], PageInfo::SUFFIX);
+				$title = basename($row['filename'], PageInfo::SUFFIX);
+			} else {
```

Comment:
```suggestion
			} elseif ($internalPath === '' || $internalPath === '.') {
				$title = $this->l10n->t('Landing page');
			} else {
```

We have to special-treat landing pages here.